config/crds/resources/apiextensions.k8s.io_v1_customresourcedefinition_vertexaiendpoints.vertexai.cnrm.cloud.google.com.yaml

```diff
@@ -162,233 +207,669 @@ spec:
                       type: string
                   type: object
                 type: array
-              createTime:
-                description: Output only. Timestamp when this Endpoint was created.
+              observedGeneration:
+                description: ObservedGeneration is the generation of the resource
+                  that was most recently observed by the Config Connector controller.
+                  If this is equal to metadata.generation, then that means that the
+                  current reported status reflects the most recent desired state of
+                  the resource.
+                type: integer
+              observedState:
+                description: The observed state of the underlying GCP resource.
+                properties:
+                  createTime:
```

Comment:
Mentioned this on the other PR, but if it's easy we should consider not including createTime / etags in observedState _yet_.  Etags in particular I worry is going to be laggy and not very useful.  createTime I guess I don't really have a concrete objection to, just a question of whether we should default-yes or default-no.

Reply:
Removing "etag" sounds good to me.
I think we shipped `VertexAIDataset` with "createTime", we could do the same for this resource.

Reply:
The concern is if anyone else writes to the object (possibly including internal clients), etags will be out of date until our next reconcile.  We don't want to encourage people to write to the object directly, and that's the primary use for etag.  etag also is intended to just be an opaque "cookie" for optimistic concurrency.

OTOH, if/when we implement the "LIST" optimization over polling each object every 10 minutes, maybe this field will be useful.

But ... it seems like we don't need it today, and we only have speculative uses for it.

Reply:
#1637